engine/src/main/java/org/destinationsol/game/screens/MapScreen.java

```diff
@@ -15,14 +15,76 @@
  */
 package org.destinationsol.game.screens;
 
-import com.badlogic.gdx.math.Rectangle;
-import org.destinationsol.GameOptions;
 import org.destinationsol.SolApplication;
 import org.destinationsol.game.MapDrawer;
-import org.destinationsol.game.SolGame;
 import org.destinationsol.ui.SolInputManager;
 import org.destinationsol.ui.SolUiBaseScreen;
+import org.destinationsol.ui.responsiveUi.UiHeadlessButton;
+import org.destinationsol.ui.responsiveUi.UiRelativeLayout;
+import org.destinationsol.ui.responsiveUi.UiTextButton;
+import org.destinationsol.ui.responsiveUi.UiVerticalListLayout;
+
+import static org.destinationsol.ui.UiDrawer.UI_POSITION_RIGHT;
 
 
 public class MapScreen extends SolUiBaseScreen {
+    private MapDrawer mapDrawer;
+    private UiTextButton cancelButton;
+    private UiTextButton zoomInButton;
+    private UiTextButton zoomOutButton;
+
+    @Override
+    public void onAdd(SolApplication solApplication) {
+        mapDrawer = solApplication.getGame().getMapDrawer();
+        mapDrawer.setToggled(true);
+        UiVerticalListLayout verticalListLayout = new UiVerticalListLayout();
+        cancelButton = new UiTextButton()
+                .setDisplayName("Cancel")
+                .setTriggerKey(solApplication.getOptions().getKeyClose())
+                .setOnReleaseAction(uiElement -> {
+                    mapDrawer.setToggled(false);
+                    SolApplication.changeScreen(solApplication.getGame().getScreens().mainGameScreen);
+                });
+        UiHeadlessButton alternateCancelKeyButton = new UiHeadlessButton()
+                .setTriggerKey(solApplication.getOptions().getKeyMap())
+                .setOnReleaseAction(uiElement -> {
+                   cancelButton.maybeFlashPressed(solApplication.getOptions().getKeyClose());
+                });
+        zoomOutButton = new UiTextButton()
+                .setDisplayName("Zoom out")
+                .setOnReleaseAction(uiElement -> {
```

Comment:
Any reason to prefer this over `.setOnReleaseAction(uiElement -> mapDrawer.changeZoom(false);` ?

(The indentation isn't too bad for this one either, so I don't understand why you didn't prefer the shorter version)